### PR TITLE
Article remove from machine translation process with the help of sparql.

### PR DESCRIPTION
**Pull request by axif0**
### Contributor checklist

- [X] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

When i try to run `Scribe-Data/src/scribe_data/language_data_extraction/English/translations$` `python3.9 translate_words.py
` It gives me this error message,

```
Traceback (most recent call last):
  File "/media/asif/Scribe-Data/src/scribe_data/language_data_extraction/English/translations/translate_words.py", line 54, in <module>
    translate_to_other_languages(
  File "/media/asif/Scribe-Data/src/scribe_data/translation/translation_utils.py", line 105, in translate_to_other_languages
    translations[word] = {}
TypeError: list indices must be integers or slices, not str
(venv) asif@asif-X450LA:/media/asif/Mahbub11/Scribe-Data/src/scribe_data/language_data_extraction/English/translations$ 
```

Therefore, after realizing this, i modify the [code](https://github.com/axif0/Scribe-Data/blob/b6b85c168bb77056ff586c952e434496b253e1a5/src/scribe_data/translation/translation_utils.py#L171) to match  [scribe_data_json_export](https://github.com/scribe-org/Scribe-Data/tree/64eb52b0be2d4fe7d9f212d733c602463c79046d/scribe_data_json_export) . 

> Qid Can be taken from `Cli_utils`, but sadly couldn't see a function. That's why I called it again. 

image

I'm extremely sorry for not testing all the languages and not even fully completely test for one language, my potato old pc couldn't handled it. :( 

I'm not sure about the Partitive Articles, In french articles, i see SPARQL query for French articles returns:
tout, le, un, l', toute, toutes, tous, la, les, une

It's missing:
1. Indefinite plural "des"
2. Partitive articles "du," "de la," "de l'"

How can I modify the query to include these compound articles? Are they represented differently in Wikidata? Or am I in the right path?

### Related issue

- #96

### CODE REVIEW

**Comment by axif0** [comment]
Hey, @andrewtavis hope you are well. Can you please check the pr if you have time?

**Comment by andrewtavis** [comment]
Hey @axif0 👋 Sorry for the wait here. I've been on and am on an extended vacation, but I'm now in the stage where I can work on some things :) Will check this out!

**Comment by andrewtavis** [comment]
> How can I modify the query to include these compound articles? Are they represented differently in Wikidata? Or am I in the right path?

I think that all's fine, @axif0! Wikidata might just not have the appropriate relationships set yet, but it'll get there 😊 You can go in and edit them yourself, if you'd like to :) Here's the LID for des: https://www.wikidata.org/wiki/Lexeme:L9111.

If you want to find lexemes to edit on Wikidata, you can do a search with `L:` at the start of it like [this search here](https://www.wikidata.org/w/index.php?go=Go&search=L%3A+des&search=L%3A+des&title=Special:Search&ns0=1&ns120=1&searchToken=ab0ikf86d0at5uhu2ewabd01i).

**Comment by andrewtavis** [comment]
The one thing that I'm seeing here that's jumping out to me is that `translations` has been switched over to a list. I'll check it out a bit more, but this should be ok as I think that a list of dictionaries for each of the language translations also makes sense 😊

File: src/scribe_data/translation/translation_utils.py [hunk 1]
```diff
@@ -24,14 +24,77 @@
...
+
+def remove_articles_from_words(
+    batch_words: List[str], articles: List[str]
+) -> List[str]:
+    """Remove articles from a given list of words."""
+
+    def remove_article(word: str) -> str:
+        for article in articles:
+            if word.lower().startswith(article.lower() + " "):
+                print(f"Article '{article}' found in word: '{word}'")
```
Comment (andrewtavis) [inline src/scribe_data/translation/translation_utils.py:92]: We don't need the print statement here for each of the instances of articles being found :)

Comment (andrewtavis) [reply src/scribe_data/translation/translation_utils.py:92]: I have this removed locally 😊

File: src/scribe_data/translation/translation_utils.py [hunk 2]
```diff
@@ -101,10 +166,19 @@ def translate_to_other_languages(source_language, word_list, translations, batch
             )
 
             for word, translation in zip(batch_words, translated_words):
-                if word not in translations:
-                    translations[word] = {}
-
-                translations[word][lang_code] = translation
+                "Find if the word already exists in translations"
```
Comment (andrewtavis) [inline src/scribe_data/translation/translation_utils.py:169]: Quick note that you have a string here instead of a comment :) Fixed locally 😊

**Review by andrewtavis — APPROVED** [review]
Thanks for all the work here, @axif0! There's a bit more work to be done, but we can finish it all up in #70 where the translations will be ran anyway 😊 Really appreciate the discussions on this and all the efforts you put in!

**Pull request merged**